drivers/serial/uart_realtek_rts5912.c

```diff
@@ -0,0 +1,382 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+ * Author: Lin Yu-Cheng <lin_yu_cheng@realtek.com>
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_uart
```

Comment:
Is the UART based on the National Semiconductor 16550 implementation?  There is a already a [NS16550 driver](https://github.com/zephyrproject-rtos/zephyr/blob/765bfbbc3c4b27ce7650e664aa04b73951e29b5f/drivers/serial/uart_ns16550.c) in the tree.

I'm aware of at least one implementation that uses the NS16550 driver, but then also creates a wrapper driver to handle vendor extensions beyond the NS16550 standard.

[ITE it8xxx2 SoC](https://github.com/zephyrproject-rtos/zephyr/blob/765bfbbc3c4b27ce7650e664aa04b73951e29b5f/dts/riscv/ite/it8xxx2.dtsi#L126) uses the NS16550 driver and provides the `ite,it8xxx2-uart` wrapper.

Reply:
Some of our register usage differs from the NS16550 driver, so we'll keep our approach in place for now.

Reply:
Please consider using the existing NS16550 driver in a future pull request.

Reply:
We will keep in mind and add it to the working list.

Reply:
note: there are a few vendor quirks in ns16550 driver

Reply:
We change uart device driver to ns16550 driver now, thanks.